pgscatalog_utils/scorefile/genome_build.py

```diff
@@ -1,55 +1,41 @@
-    """ Annotate the dataframe with genome build data """
+    """ Annotate the dataframe with genome build data  """
     logger.debug(f"Annotating target build: {target_build}")
     build_dict: dict = {'GRCh37': 'hg19', 'GRCh38': 'hg38', 'hg19': 'hg19', 'hg38': 'hg38'}  # standardise build names
-    df['target_build'] = build_dict[target_build]
-
-    builds: pd.DataFrame = _get_builds(df['filename'].drop_duplicates())
-    builds['genome_build'] = builds.apply(lambda x: build_dict[x.genome_build], axis=1)
-    return df.merge(builds, how="left", on="filename")
-
-
-def _read_header(f: TextIO) -> str:
-    """ Extract genome build of scorefile from PGS Catalog header format """
-    for line in f:
-        if re.search("^#genome_build", line):
-            # get #genome_build=GRCh37 from header
-            header = line.replace('\n', '').replace('#', '').split('=')
-            # and remap to liftover style
-            try:
-                build: str = header[-1]
-                logger.debug(f"Valid genome build detected: {build}")
-                return build
-            except KeyError:
-                raise Exception("Bad genome build detected in header")
-        elif line[0] != '#':
-            raise Exception("No genome build detected in header")
+    df['chain_target_build'] = build_dict[target_build]
+    df = df.assign(chain_genome_build=[build_dict[x] for x in df['genome_build']])
+    return df
+
+def build2GRC(build):
+    """Map build names so they can be compared with GRCh37 and 38"""
+    build_2_GRC_dict = {'GRCh37': 'GRCh37', 'GRCh38': 'GRCh38', 'hg19': 'GRCh37', 'hg38': 'GRCh38'}  # standardise build names
+    if pd.isnull(build):
+        return None
+    else:
+        return build_2_GRC_dict.get(build)
+
 
 
 def _read_build(path: str) -> str:
     """ Open scorefiles and automatically handle compressed input """
     logger.debug(f'Reading header of {path}')
-    try:
-        with io.TextIOWrapper(gzip.open(path, 'r')) as f:
-            return _read_header(f)
-    except gzip.BadGzipFile:
-        with open(path, 'r') as f:
-            return _read_header(f)
+    h = _read_header(path)
+    return {k: h.get(k, None) for k in ('genome_build', 'HmPOS_build')}
 
 
-def _get_builds(s: pd.Series) -> pd.DataFrame:
+def _get_builds(paths: list) -> pd.DataFrame:
     """ Get genome builds for a series of scorefile paths
-        | filename | -> | filename | genome_build |
-        | x.txt.gz |    | x.txt.gz | hg19         |
+        | filename              | -> |                       | genome_build | HmPOS_build |
+        | x.txt.gz              |    | x.txt.gz              | hg19         | None        |
+        | x_hmPOS_GRCh37.txt.gz |    | x_hmPOS_GRCh37.txt.gz | hg19         | GRCh37      |
     """
-    return pd.concat([s, s.apply(_read_build).rename("genome_build")], axis=1)
+    return pd.DataFrame.from_dict({path: _read_build(path) for path in paths}, orient='index')
+
```

Comment:
This function is defined but never called

I think both `_read_build` and `_get_builds` can be deleted with the new main `combine_scorefiles()` function taking over

Reply:
Yes, I think so! I left them in mostly for legacy reasons in case we wanted to refactor it to do it closer to the old way. But I think the current way makes more sense/is more controllable